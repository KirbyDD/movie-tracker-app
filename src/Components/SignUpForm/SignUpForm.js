import React, { Component} from 'react';
import SignIn from '../../Containers/SignIn/SignIn';
import SignUp from '../../Containers/SignUp/SignUp';

class SignUpForm extends Component {
  constructor() {
    super();
    this.state = {
      signIn: false,
      signUp: false,
    }
  }

  toggleSignInAndUp = (e) => {
    e.target.name === 'sign-in' ?
    this.setState({signIn: true, signUp: false}) :
    this.setState({signIn: false, signUp: true})
  }

  render() {
    
  return (
    <section>
      <div>
        <button onClick={this.toggleSignInAndUp} name="sign-in">Sign In</button>
        <button onClick={this.toggleSignInAndUp} name="sign-up">Sign Up</button>
      </div>
      {this.state.signIn && <SignIn />}
      {this.state.signUp && <SignUp />}
    </section>
  )
  }
}



export default SignUpForm;
