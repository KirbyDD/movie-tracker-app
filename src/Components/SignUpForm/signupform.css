.switch-to-signup-btn {
  position: absolute;
  right: 32px;
  top: 16px;
  cursor: pointer;
  background: transparent;
  color: #fff;
  border: none;
  outline: none;
  font-size: 16px;
}

.switch-to-signup-btn:hover {
  color: #01D277;
}
