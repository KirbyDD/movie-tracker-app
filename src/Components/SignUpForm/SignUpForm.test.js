import React from 'react';
import {shallow} from 'enzyme';
import { SignUpForm } from './SignUpForm';

describe('SignUpForm', () => {
  let wrapper;
  beforeEach(function() {
    wrapper = shallow(
      <SignUpForm />
    )
  })

  it('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should have default properties', () => {
    expect(wrapper.state().signIn).toEqual(true);
    expect(wrapper.state().signUp).toEqual(false);
  })

  it('Should toggle states when button is clicked', async () => {
   await  wrapper.find('button').simulate('click', {
      target: {name: 'sign-up'}
    })
    expect(wrapper.state().signIn).toEqual(false);
    expect(wrapper.state().signUp).toEqual(true);
    await  wrapper.find('button').simulate('click', {
      target: {name: 'sign-in'}
    })
    expect(wrapper.state().signIn).toEqual(true);
    expect(wrapper.state().signUp).toEqual(false);
  })
})