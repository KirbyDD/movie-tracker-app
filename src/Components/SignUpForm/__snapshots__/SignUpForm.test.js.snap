// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm Should match snapshot 1`] = `
<section>
  <button
    className="switch-to-signup-btn"
    name="sign-up"
    onClick={[Function]}
  >
    Sign Up
  </button>
  <Connect(SignIn) />
</section>
`;
